Fast math

adpaco-aws opened the pull request:
### Description of changes: 

Models fast math intrinsics as regular floating point operations. Includes function to check inputs are finite arguments. Adds 10 new test cases (5 positive, 5 negative) for testing fast math intrinsics.

### Resolved issues:

Resolves #788 


### Call-outs:

<!-- 
Address any potentially confusing code. Is there code added that needs to be cleaned up later? Is there code that is missing because it’s still in development? 
-->

### Testing:

* How is this change tested? Existing regression + 10 new tests.

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

adpaco-aws commented on src/kani-compiler/cbmc/src/goto_program/stmt.rs:
```diff
@@ -129,6 +129,13 @@ impl Stmt {
         }
     }
+
+    pub fn get_stmts(&self) -> Option<&Vec<Stmt>> {
```
This is part of an earlier version where the intrinsics where encoded in a block statement, so I needed to retrieve statements within in order to prepend the finite checks. This version does not use it, but I figured it could be useful for something else. We can remove it otherwise.

celinval replied:
If you do keep it, can you please add a comment?

adpaco-aws replied:
Added a comment.

celinval commented on tests/kani/Intrinsics/FastMath/add.rs:
```diff
@@ -0,0 +1,19 @@
+        (false, false) => kani::assume(x > f64::MIN - y),
+        _ => (),
+    }
+    let _z = unsafe { std::intrinsics::fadd_fast(x, y) };
```
Can you please assert that the result is equal to regular add? The same for the other tests.

adpaco-aws replied:
Changed it so we assert it in the case of addition and substraction, as other operations show serious performance issues. Created #809 to revisit this.

celinval reviewed (commented):
Nice test coverage! Just minor comments.

adpaco-aws commented:
Some changes I made:
 1. Added tests for `f32` values so we now cover all floating point types.
 2. Changed tests to assert equality (w.r.t. regular operations) in the case of addition and substraction, but other operations show serious performance issues. Created #809 to revisit this topic later.
 3. Dropped support for `frem_fast` because testing showed unexpected results. Created #810 to explain the issue and tackle it in the future.
